accord-core/src/main/java/accord/impl/progresslog/DefaultProgressLog.java

```diff
@@ -260,7 +227,54 @@ else if (after.durability().isDurableOrInvalidated() && (force || !before.durabi
         }
     }
 
-    private TxnState maybeFetch(SafeCommandStore safeStore, TxnId txnId, Command after, TxnState state)
+    private TxnState updateHomeState(SafeCommandStore safeStore, Command after, @Nullable TxnState state)
+    {
+        Route<?> route = after.route();
```

Comment:
I personally would annotate this method. It might be self-evident, but adding some descriptions of the branches, especially since we're querying some conditions more than once, could be beneficial.